tests/qos/test_pfc_pause.py

```diff
@@ -88,31 +136,37 @@ def run_test_t0(fanouthosts,
         if send_pause:
             peer_device = conn_graph_facts['device_conn'][duthost.hostname][dut_intf_paused]['peerdevice']
             peer_port = conn_graph_facts['device_conn'][duthost.hostname][dut_intf_paused]['peerport']
-            peer_port_name = eos_to_linux_intf(peer_port)
-            peerdev_ans = fanouthosts[peer_device]
+            peer_info = { 'peerdevice': peer_device,
+                          'pfc_fanout_interface': peer_port
+                        }
 
             if not pfc_pause:
                 pause_prio = None
 
-            start_pause(host_ans=peerdev_ans,
-                        pkt_gen_path=PFC_GEN_REMOTE_PATH,
-                        intf=peer_port_name,
-                        pkt_count=PFC_PKT_COUNT,
-                        pause_duration=pause_time,
-                        pause_priority=pause_prio)
-
+            if i == 0:
```

Comment:
should it be "if not strom_handle:"?

Reply:
Looks like you have everything except peer_info outside the loop. Can you create thsi storm_handle outside the loop?

Reply:
Storm handle cannot be created outside since it needs peer_info which is a mandatory argument for that class init. I only get that info inside the loop. For each iteration, it picks up a different set of interfaces and then I need to update the peer_info accordingly so that I have the handle to the correct fanout.

Reply:
The peer info is updated at line 152. It might be better to make peer_info optional then.

Reply:
update_peer_info() method was added to allow updating fanout info/interfaces during the test run without having to init the class again (which might be needed in some testcases where we are trying to generate storm one after the other on different interfaces)
If I make peer info optional, then the caller has to make sure that this method is always called even if the test is being run against a single interface. Also I have some checks in the class init to validate these args are passed in.
what are your thoughts?

Reply:
@neethajohn  please change i == 0 check to not storm_handle to be more generic.

Reply:
Done!